apps/voting/contracts/Voting.sol

```diff
@@ -21,29 +21,29 @@ contract Voting is IForwarder, AragonApp {
     bytes32 public constant MODIFY_SUPPORT_ROLE = keccak256("MODIFY_SUPPORT_ROLE");
     bytes32 public constant MODIFY_QUORUM_ROLE = keccak256("MODIFY_QUORUM_ROLE");
 
-    uint256 public constant PCT_BASE = 10 ** 18; // 0% = 0; 1% = 10^16; 100% = 10^18
+    uint64 public constant PCT_BASE = 10 ** 18; // 0% = 0; 1% = 10^16; 100% = 10^18
 
     enum VoterState { Absent, Yea, Nay }
 
     struct Vote {
         address creator;
+        bool executed;
         uint64 startDate;
-        uint256 snapshotBlock;
-        uint256 supportRequiredPct;
-        uint256 minAcceptQuorumPct;
+        uint64 snapshotBlock;
+        uint64 supportRequiredPct;
+        uint64 minAcceptQuorumPct;
```

Comment:
Worried about these being uint64 given that the max uint64 is `~1.8 * 10^19`and the max values they can have is `10^18`. We should at least explicitly cast `uint64` to `uint256` whenever used for math operations in `canExecute` and `isValuePct`.

Also extending these tests [here](https://github.com/aragon/aragon-apps/blob/ba4538d1e49cc36d411bc7bbd103c3348ed8c95d/apps/voting/test/voting.js#L151) so we also verify the math still works for bigger token amounts.

Reply:
Sure thing, I went through and made sure they were explicitly converted before any math was performed.

I'm having trouble understanding how I would change those tests, they seem to test for 18 decimals already it seems. What should I change?

(sorry for the duplicate messages, I clicked the wrong comment button)

Reply:
I would just add an extra decimal amount to test, so it is `[0, 2, 18, 26]` just to be extra careful

Reply:
KK 👍